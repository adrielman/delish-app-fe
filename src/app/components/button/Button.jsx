import React from "react";
import styles from "./Button.module.scss";

export const Button = ({ onClick, text }) => {
  const clickHandler = event => {
    event.preventDefault();
    onClick();
  };
  return (
    <button className={styles.button} onClick={clickHandler}>
      {text}
    </button>
  );
};
