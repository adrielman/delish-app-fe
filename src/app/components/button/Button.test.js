import React from "react";
import { shallow } from "enzyme";

import Button from "./Button";

describe("Button", () => {
  it("should indeed render correctly", () => {
    const component = shallow(
      <Button
        onClick={() => {
          console.log("I work!");
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
