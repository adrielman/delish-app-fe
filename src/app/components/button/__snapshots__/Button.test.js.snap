// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should indeed render correctly 1`] = `
<button
  className="button"
  onClick={[Function]}
/>
`;
